### PR TITLE
Fix #384 - provide Docker image

### LINKED ISSUE #384: [REQUEST] Specify a reference Dockerfile which is tested in CI
**Is your feature request related to a problem? Please describe.**
I was frustrated with getting the validator up and running. 
I'd like to have a reproducible environment specified that "just works". 

**Describe the solution you'd like**
1. A [Dockerfile](https://docs.docker.com/engine/reference/builder/) for a reference environment which is tested in the CI suite.
2. A [docker-compose](https://docs.docker.com/compose/) with setup for the simple UI 

**Describe alternatives you've considered**
- More step by step instructions with special instructions for every hardware/software/data combination issue
- More time spent troubleshooting individual hardware situations

**Additional context**
Some discussion in : https://github.com/MobilityData/gtfs-validator/issues/383

### LINKED ISSUE #405: [REQUEST] Add functional tests to testing suite
**Is your feature request related to a problem? Please describe.**
As a local user, I would appreciate end-to-end functional tests be included in *repository* testing suite (as opposed to in a GitHub-triggered-action) in order to make sure the code *in my environment* is working appropriately from end-to-end. 

**Describe the solution you'd like**
A set of end-to-end functional tests for a variety of feed sources and settings run within local tests. 
https://guides.gradle.org/testing-gradle-plugins/ has some suggestions. 

**Describe alternatives you've considered**
While the GitHub triggered functional test is useful, it is not easily run locally by non-contributors *and* isn't necessarily run as a matter of course on local machines prior to pushing code.

### PR DESCRIPTION
**Pull request by fabrice-v**
**Summary:**
closes #384 
closes #405 

This PR contains changes so that a Docker image is built and store in GitHub Container Repository: https://github.com/orgs/MobilityData/packages/container/package/gtfs-validator

**Expected behavior:** 

When checks are run, a docker image should be built and stored in the GitHub Container Registry

Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Run the unit tests with `gradle test` to make sure you didn't break anything
- [x] Format the title like "Fix #<issue_number> - <short description of fix and changes>" (for example - "Fix #1111 - Check for null value before using field")
- [x] Linked all relevant issues
- [ ] Include screenshot(s) showing how this pull request works and fixes the issue(s)

### CODE REVIEW

**Comment by fabrice-v** [comment]
@lionel-nj , this is now ready for review but will not be merged before #365 is on master (I will fix conflicts in this branch)

File: .github/workflows/docker.yml [hunk 1]
```diff
@@ -0,0 +1,62 @@
+name: Docker image
+
+#on:
```
Comment (lionel-nj) [inline .github/workflows/docker.yml:3]: Could this be removed?

**Review by lionel-nj — APPROVED** [review]
LGTM after removing the comments in docker.yml

File: .github/workflows/docker.yml (hunk 1 above)
Comment (fabrice-v) [reply .github/workflows/docker.yml:3]: the intent was to keep it a a reference should we decide to provide images of our releases only. I could clarify that with another comment?

**Comment by fabrice-v** [comment]
YES! Branch is fixed
TODOS

- [ ] add war to Docker image
- [ ] add documentation regarding Docker image usage
- [ ] reorder usage in README
- [ ] add war packaging and JS tests to Circle CI

**Review by lionel-nj — APPROVED** [review]
As discussed, there is still:
- [ ] provide a second docker image
- [ ] update documentation

**Comment by fabrice-v** [comment]
Reference for docker workflow file https://github.com/cicirello/alpine-plus-plus/blob/master/.github/workflows/docker-publish.yml

**Pull request merged**